blog/_posts/2025-03-17-enhancing-observability-in-azure-kubernetes-service.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Enhancing Observability in Azure Kubernetes Service (AKS): What's New?"
+description: "Recapping some of the recent announcements and improvements in AKS observability experience."
+date: 2025-03-17
+authors: Pavneet Ahluwalia
+categories: observability
+tags:
+  - azure-monitor
+  - observability
+  - metrics
+---
+
+# Observability Enhancements in Azure Kubernetes Service (AKS)
+
+At Azure Kubernetes Service (AKS), we deeply recognize how crucial observability is for running stable Kubernetes environments. Given our extensive reliance on Kubernetes internally, we're continually innovating to ensure you have robust, clear, and actionable insights into your cluster health and performance. Observability—the ability to monitor, understand, and manage your systems effectively—is a foundational pillar for AKS product vision to enable our users to achieve more.
+
+In this post, we'll highlight several recent observability enhancements in AKS and Azure Monitor across three main dimensions: Improved monitoring capabilities, Reduce effort for best-in-class observability, and better reliability and performance.
+
+## Improced Monitoring Capabilities
+
+### 1. Preview: AKS Control Plane Platform Metrics
+
+AKS provides a managed control plane experience from provisioning, patching, to managed upgrades. However, at high scale, monitoring the control plane of your AKS cluster for latency, rogue call patterns and limits is important for maintaining stability and performance. AKS already supported detailed control plane metrics through Azure Monitor managed service for Prometheus. Recently, we've introduced [Azure Platform Metrics for AKS Control Plane Monitoring](https://techcommunity.microsoft.com/blog/appsonazureblog/azure-platform-metrics-for-aks-control-plane-monitoring/4385770), these free metrics deliver key insights into the utilization of your AKS control plane and understand how close to the limits you are flying.
+
+#### Key Benefits:
+- **Monitor important metrics** such as kube-apiserver CPU and Memory utilization and etcd database utilization.
+- **Anticipate and plan for scaling needs** by clearly understanding your cluster’s control plane usage and limits.
+- **Mitigate potential downtime risks** through proactive alerting on Apiserver memory usage or etCD database usage.
+
+#### How to Enable/Use:
+- These platform metrics are available by default.
+- To view these metrics, sign in to the Azure Portal and navigate directly to your AKS cluster.
+- From the left navigation menu, select **Monitoring**, then click **Metrics**.
+- Under the "Container Service" Namespace, chose metric eg: API server memory percentage
+![Control plane Metrics!](https://github.com/pavneeta/AKS/blob/obs_blog/blog/assets/images/enhancing-observability/Control-plane-metrics.png)
```

Comment:
To render images, just use a relative path starting from the root of the rendered site on GitHub pages, which is `/AKS`

```suggestion
![Control plane Metrics!](https://github.com/pavneeta/AKS/blob/obs_blog/blog/assets/images/enhancing-observability/Control-plane-metrics.png)
```
```suggestion
![Control plane Metrics!](/AKS/assets/images/enhancing-observability/Control-plane-metrics.png)
```